GetIntoTeachingApi/Services/CrmService.cs

```diff
@@ -222,14 +222,19 @@ public IEnumerable<TeachingEvent> GetTeachingEvents(DateTime? startAfter = null)
             query.Criteria.AddFilter(filter);
 
             var link = query.AddLink("msevtmgt_building", "msevtmgt_building", "msevtmgt_buildingid", JoinOperator.LeftOuter);

```

Comment:
The entire `link` should be dropped here as well, you've just removed the columns setting for the link with this change. Did you check if GetTeachingEvents is used elsewhere? I don't think it is, but if it is we may need to include buildings for other scenarios and exclude for the sync scenario.